lib/virt_autotest/utils.pm

```diff
@@ -55,8 +56,22 @@ sub restart_libvirtd {
     else {
         systemctl("restart libvirtd", timeout => 180);
     }
+    record_info("Libvirtd has been restarted!");
     save_screenshot;
-    record_info("Debug log for libvirtd has been enabled!");
+}
+
+# usage: restart_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be restarted: restart_modular_libvirt_daemons(virtqemud virtstoraged ...)
+# to restart all modular daemons without any daemons passed
+sub restart_modular_libvirt_daemons {
+    my @daemons = @_ == 0 ? qw(virtqemud virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd) : split(" ", @_);
+    if (is_alp) {
+        record_soft_failure("Restaring modular libvirt daemons has not been emplemented in ALP. See poo#129086");
+    }
+    else {
+        systemctl("restart $_\{,-ro,-admin\}.socket") foreach @daemons;
```

Comment:
Would you please also add  "restart $_.service" ? Sorry if you were misled by my wip modular libvirt confluence page, which didn't differentiate clearly for switching to modular and restart of modular.

Reply:
Done.